packages/runtime/container-runtime/src/summaryManager.ts

```diff
@@ -251,30 +259,49 @@ export class SummaryManager extends TypedEventEmitter<ISummaryManagerEvents> imp
         }
     }
 
-    private checkBypassInitialDelay() {
-        if (!this.initialDelay.isCompleted && this.summaryCollection.opsSinceLastAck >= this.opsToBypassInitialDelay) {
-            this.initialDelay.resolve();
+    /**
+     * Implements initial delay before creating summarizer
+     * @returns true, if creation is delayed due to heuristics (not many ops to summarize).
+     *          False if summarizer should start immediately due to too many unsummarized ops.
+     */
+    private async delayBeforeCreatingSummarizer(): Promise<boolean> {
+        // throttle creation of new summarizer containers to prevent spamming the server with websocket connections
+        let delayMs = this.startThrottler.getDelay();
+        if (delayMs > 0 && delayMs >= this.startThrottler.maxDelayMs) {
+            // we can't create a summarizer for some reason; raise error on container
+            this.emit(
+                "summarizerWarning",
+                createSummarizingWarning("SummaryManager: CreateSummarizer Max Throttle Delay", false),
+            );
         }
-    }
 
-    private async createSummarizer(delayMs: number): Promise<ISummarizer> {
         // We have been elected the summarizer. Some day we may be able to summarize with a live document but for
         // now we play it safe and launch a second copy.
         this.logger.sendTelemetryEvent({
             eventName: "CreatingSummarizer",
-            delayMs,
+            throttlerDelay: delayMs,
+            initialDelay: this.initialDelayMs,
             opsSinceLastAck: this.summaryCollection.opsSinceLastAck,
+            opsToBypassInitialDelay: this.opsToBypassInitialDelay,
         });
 
-        const shouldDelay = delayMs > 0;
-        if (shouldDelay || !this.initialDelay.isCompleted) {
-            await Promise.all([
-                this.initialDelay.promise,
-                shouldDelay ? delay(delayMs) : Promise.resolve(),
-            ]);
+        // This delay helps ensure that last summarizer that might be left from previous client
+        // has enough time to complete its summary and thus new summarizer not conflict with previous one.
+        // A better design would be for summarizer election logic to always select current summarizer as
+        // summarizing client (i.e. clientType === "summarizer" can be elected) to ensure that nobody else can
+        // summarizer while it finishes its work and moves to exit.
+        // It also helps with pure boot scenario (single client) to offset expensive work a bit out from
+        // critical boot sequence.
+        let result = false;
```

Comment:
nit: Consider renaming this to something like `startImmediately` or something that conveys what this means. It's not evident what is happening here and how this should be used.